lib/driver.js

```diff
@@ -309,6 +309,18 @@ class XCUITestDriver extends BaseDriver {
     }
   }
 
+  /**
+   * Returns the default URL for Safari browser
+   * @returns {string} The default URL
+   */
+  getDefaultUrl () {
+    return this.isRealDevice()
+      ? 'http://appium.io'
+      : `http://${
+        this.opts.address.includes(':') ? `[${this.opts.address}]` : this.opts.address
```

Comment:
Would it make sense to use some more sophisticated ipv6 address checks like https://www.npmjs.com/package/ip-regex ?

Reply:
I initially thought to use something like the package, but at this time, I haven't since the main use case should be a local address. No strict check should be enough then.